rasa/server.py

```diff
@@ -1438,12 +1440,20 @@ def _test_data_file_from_payload(
 ) -> Text:
     if request.headers.get("Content-type") == YAML_CONTENT_TYPE:
         return str(
-            _training_payload_from_yaml(request, temporary_directory)["training_files"]
+            _training_payload_from_yaml(
+                request,
+                temporary_directory,
+                # test stories have to prefixed with `test_`
+                file_name=f"{TEST_STORIES_FILE_PREFIX}data.yml",
```

Comment:
should all this new server behaviour be described in the changelog?

Reply:
this doesn't change any server behavior. It's rather needed because our `test` functions now enforce our rules for test files (e.g. that they have to start with `test_`)

Reply:
So it's now writing a different file name than before? That seems like a change in behaviour to me!

Reply:
Not for the user. They send a file via API and we save it to a temporary directory before loading it for testing. How we name it and how the temporary directory looks like doesn't affect the user and is purely internal in my opinion.

Reply:
ah ok gotcha! 👍